packages/asset-ownership-service/src/services/substream.ts

```diff
@@ -0,0 +1,537 @@
+                      const key = addressTableLookup.accountKey.toBase58();
+                      if (!lookupTableCache[key]) {
+                        const lookupTableResult =
+                          await provider.connection?.getAddressLookupTable(
+                            addressTableLookup.accountKey
+                          );
+                        if (lookupTableResult?.value) {
+                          lookupTableCache[key] = lookupTableResult.value;
+                        }
+                      }
+                      if (lookupTableCache[key]) {
+                        addressLookupTableAccounts.push(lookupTableCache[key]);
+                      }
+                    }
+
+                    const { staticAccountKeys, accountKeysFromLookups } =
+                      message.getAccountKeys({
+                        addressLookupTableAccounts,
+                      });
+
+                    const accountKeys = [
+                      ...staticAccountKeys,
+                      ...(accountKeysFromLookups?.writable || []),
+                      ...(accountKeysFromLookups?.readonly || []),
+                    ];
+
+                    for (const compiledInstruction of message.compiledInstructions) {
+                      const programIdIndex = compiledInstruction.programIdIndex;
+                      const programId = new PublicKey(
+                        accountKeys[programIdIndex]
+                      );
+
+                      const instructionCoder = coders[programId.toBase58()];
+
+                      if (!instructionCoder) continue;
+
+                      const decodedInstruction = instructionCoder.decode(
+                        Buffer.from(compiledInstruction.data)
+                      );
+
+                      if (!decodedInstruction) continue;
+
+                      const formattedInstruction = instructionCoder.format(
+                        decodedInstruction,
+                        compiledInstruction.accountKeyIndexes.map((idx) => ({
+                          pubkey: new PublicKey(accountKeys[idx]),
+                          isSigner: message.isAccountSigner(idx),
+                          isWritable: message.isAccountWritable(idx),
+                        }))
+                      );
+
+                      if (!formattedInstruction) continue;
+
+                      const accountMap = Object.fromEntries(
+                        (formattedInstruction.accounts || []).map((acc) => [
+                          acc.name,
+                          acc,
+                        ])
+                      );
+
+                      const argMap = Object.fromEntries(
+                        (formattedInstruction.args || []).map((arg) => [
+                          arg.name,
+                          arg,
+                        ])
+                      );
+
+                      switch (decodedInstruction.name) {
+                        case "update_maker_tree_v0":
```

Comment:
since the create_tree is done via CPI it will be nested in the `update_maker_tree_v0` and we will need to use the hem coder to decode it. This is the same for `MintToCollectionV` and `issue_entity_v0`